libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/contract/ContainerVisibilityContract.kt

```diff
@@ -0,0 +1,91 @@
+package com.mapbox.navigation.dropin.contract
+
+import com.mapbox.navigation.dropin.processor.ContainerVisibilityProcessor
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+sealed class ContainerVisibilityAction {
+    object ForEmpty : ContainerVisibilityAction()
+    object ForFreeDrive : ContainerVisibilityAction()
+    object ForRoutePreview : ContainerVisibilityAction()
+    object ForActiveNavigation : ContainerVisibilityAction()
+    object ForArrival : ContainerVisibilityAction()
+}
+
+sealed interface ContainerVisibilityResult {
+    val volumeContainerVisible: Boolean
+    val recenterContainerVisible: Boolean
+    val maneuverContainerVisible: Boolean
+    val infoPanelContainerVisible: Boolean
+    val speedLimitContainerVisible: Boolean
+    val routeOverviewContainerVisible: Boolean
```

Comment:
It seems the way to add a view to this container in a consistent way, would be to add a new boolean visibility flag. For example, when we add the current-road-label - does it go here?

I think I prefer independent streams for visibility. You will also have independent streams for providing data to the containers, and those streams will impact the visibility. For example, speed limit my be visible to the container but is invisible when the road does not have a speed limit value.

I can see this is only connecting, NavigationState visibility. But the streams that control visibility can easily depend on NavigationState rather than defining a new action type

Reply:
Here's another idea

``` kotlin
    val speedLimitVisibilitySet = setOf(NavigationState.FreeDrive, NavigationState.RoutePreview)
    fun speedLimitVisible(): Flow<Boolean> = navigationState().map {
        speedLimitVisibilitySet.contains(it)
    }
```

Reply:
I like the suggestion. I am going to create a new issue for it where I am gonna map the view visibility state to navigation state. Let's not block this PR, as what you are suggesting is an improvement and a good one but the PR unblocks us from moving further on action items.